esp-hal-common/src/interrupt/mod.rs

```diff
@@ -1,4 +1,134 @@
 //! Interrupt support
+//!
+//! ## Overview
+//! The `interrupt` driver is a crucial module for ESP chips. Its primary
+//! purpose is to manage and handle interrupts, which are asynchronous events
+//! requiring immediate attention from the CPU. Interrupts are essential in
+//! various applications, such as real-time tasks, I/O communications, and
+//! handling external events like hardware signals.
+//!
+//! The core functionality of the `interrupt` driver revolves around the
+//! management of interrupts. When an interrupt occurs, it temporarily stops the
+//! ongoing CPU operations, saves its current state, and starts executing the
+//! corresponding interrupt service routine (ISR). The interrupt service routine
+//! is a user-defined function that performs the necessary actions to handle the
+//! specific interrupt. Once the ISR is executed, the driver restores the saved
+//! CPU state and resumes normal program execution.
+//!
+//! In scenarios where multiple interrupts may occur simultaneously, the
+//! interrupt driver determines the `priority` of each interrupt. This
+//! prioritization ensures that critical or high-priority tasks are handled
+//! first. It helps prevent delays in time-sensitive applications and allows the
+//! system to allocate resources efficiently. This functionality is provided and
+//! implemented by the `priority` enum.
+//!
+//!
+//! ## Example
+//! ```no_run
+//! #[entry]
+//! fn main() -> ! {
```

Comment:
This example seems _way_ more complicated than it needs to be. Why do we need 4 interrupt handlers?